app/controllers/api/v2/courses_controller.rb

```diff
@@ -37,20 +43,7 @@ def publish
         if @course.publishable?
           @course.publish_sites
           @course.publish_enrichment(@current_user)
-
-          response = ManageCoursesAPIService::Request.sync_course_with_search_and_compare(
-            @current_user.email,
-            @provider.provider_code,
-            @course.course_code
-          )
-
-          if response
-            head :ok
-          else
-            raise RuntimeError.new(
-              'error received when syncing with search and compare'
-            )
-          end
+          sync_with_search_and_compare
```

Comment:
`sync_with_search_and_compare` is an action according to `routes.rb`. I don't think we should be calling an action from another action like this, so `sync_with_search_and_compare` is still being used then we should be using whatever it uses to sync ... either `SearchAndCompareAPIService::Request.sync` or a new, `private` method in this controller.

Reply:
Okay. Have addressed this and the below comment. Let me know if you're happy with the solution.